static/app/components/events/interfaces/spans/spanGroupBar.tsx

```diff
@@ -0,0 +1,430 @@
+                          </SpanGroupRowTitleContent>
+                        </RowTitle>
+                      </RowTitleContainer>
+                    </RowCell>
+                    <DividerContainer>
+                      {this.renderDivider(dividerHandlerChildrenProps)}
+                    </DividerContainer>
+                    <RowCell
+                      data-type="span-row-cell"
+                      showStriping={spanNumber % 2 !== 0}
+                      style={{
+                        width: `calc(${toPercent(1 - dividerPosition)} - 0.5px)`,
+                      }}
+                      onClick={() => {
+                        toggleSpanGroup();
+                      }}
+                    >
+                      <RowRectangle
+                        spanBarHatch={false}
+                        style={{
+                          backgroundColor: theme.blue300,
+                          left: `min(${toPercent(bounds.left || 0)}, calc(100% - 1px))`,
+                          width: toPercent(bounds.width || 0),
+                        }}
+                      >
+                        <DurationPill
+                          durationDisplay={durationDisplay}
+                          showDetail={false}
+                          spanBarHatch={false}
+                        >
+                          {durationString}
+                        </DurationPill>
+                      </RowRectangle>
+                      {this.renderMeasurements()}
+                      <SpanBarCursorGuide />
+                    </RowCell>
+                    <DividerLineGhostContainer
+                      style={{
+                        width: `calc(${toPercent(dividerPosition)} + 0.5px)`,
+                        display: 'none',
+                      }}
+                    >
+                      <DividerLine
+                        ref={addGhostDividerLineRef()}
+                        style={{
+                          right: 0,
+                        }}
+                        className="hovering"
+                        onClick={event => {
+                          // the ghost divider line should not be interactive.
+                          // we prevent the propagation of the clicks from this component to prevent
+                          // the span detail from being opened.
+                          event.stopPropagation();
+                        }}
+                      />
+                    </DividerLineGhostContainer>
+                  </RowCellContainer>
+                </Row>
+              );
+            }}
+          </DividerHandlerManager.Consumer>
+        )}
+      </ScrollbarManager.Consumer>
+    );
+  }
+}
+
+export function isCollapsedSpanGroup({
+  spanGrouping,
+  showSpanGroup,
+  toggleSpanGroup,
+}: SpanGroupProps) {
+  return (
+    Array.isArray(spanGrouping) &&
+    spanGrouping.length !== 0 &&
```

Comment:
I noticed that a collapsed span group with `spanGrouping.length` === 1 takes about the same space as just expanding the group (since length === 1). Would it be better to just display the uncollapsed view in these cases?

Reply:
I've amended to not collapse the group if the collapsed group only has one span.

Reply:
👍 nice!